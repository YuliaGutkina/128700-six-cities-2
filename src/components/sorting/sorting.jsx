import React from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {ActionCreator} from "../../reducer/data/data";
import {getSortingOrder} from "../../reducer/data/selectors";
import {Select} from "../select/select";


const SORTING_OPTIONS = [
  {value: `popular`, text: `Popular`},
  {value: `to-high`, text: `Price: low to high`},
  {value: `to-low`, text: `Price: high to low`},
  {value: `top-rated`, text: `Top rated first`}
];

const Sorting = (props) => {
  const {onChangeSorting, sortingOrder} = props;

  return <form className="places__sorting" action="#" method="get" onSubmit={(e) => {
    e.preventDefault();
  }}>
    <span className="places__sorting-caption">Sort by</span>
    <Select
      onChange={onChangeSorting}
      value={sortingOrder}
      options={SORTING_OPTIONS}
      id="places-sorting"
      className="places__sorting-type"
      optionClassName="places__option"
    />
  </form>;
};

Sorting.propTypes = {
  onChangeSorting: PropTypes.func.isRequired,
  sortingOrder: PropTypes.oneOf([`popular`, `to-high`, `to-low`, `top-rated`]),
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  sortingOrder: getSortingOrder(state)
});

const mapDispatchToProps = (dispatch) => ({
  onChangeSorting: (sortBy) => {
    dispatch(ActionCreator.changeSorting(sortBy));
  },
});


export {Sorting};
export default connect(mapStateToProps, mapDispatchToProps)(Sorting);
