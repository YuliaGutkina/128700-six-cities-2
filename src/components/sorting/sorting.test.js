import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {Sorting} from "./sorting";


it(`Sorting correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <Sorting
        sortingOrder="to-high"
        onChangeSorting={jest.fn()}
      />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
