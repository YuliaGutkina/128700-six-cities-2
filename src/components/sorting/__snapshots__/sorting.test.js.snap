// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting correctly renders after relaunch 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
  onSubmit={[Function]}
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <Select
    className="places__sorting-type"
    id="places-sorting"
    onChange={[MockFunction]}
    optionClassName="places__option"
    options={
      Array [
        Object {
          "text": "Popular",
          "value": "popular",
        },
        Object {
          "text": "Price: low to high",
          "value": "to-high",
        },
        Object {
          "text": "Price: high to low",
          "value": "to-low",
        },
        Object {
          "text": "Top rated first",
          "value": "top-rated",
        },
      ]
    }
    value="to-high"
  />
</form>
`;
